daemon/daemon.go

```diff
@@ -407,20 +407,12 @@ func (daemon *Daemon) reserveName(id, name string) (string, error) {
 
 		conflictingContainer, err := daemon.GetByName(name)
 		if err != nil {
-			if strings.Contains(err.Error(), "Could not find entity") {
-				return "", err
-			}
```

Comment:
@mountkin hm, good question w.r.t the graph.db

> If we only delete the index and config of the container when the user run docker rm xxx, there'll be garbage data left on the disk.

But that would be the same as the "automagic" removal?